agent-control/src/agent_control/run.rs

```diff
@@ -160,10 +158,10 @@ impl AgentControlRunner {
         let signature_validator = config
             .opamp
             .map(|fleet_config| {
-                build_signature_validator(fleet_config.signature_validation, config.proxy)
+                SignatureValidator::new(fleet_config.signature_validation, config.proxy)
             })
             .transpose()?
-            .unwrap_or(SignatureValidator::Noop);
+            .unwrap_or(SignatureValidator::new_noop());
```

Comment:
We also need the the Noop thing here because we need to build a signature validator even if OpAMP is disabled, right? 🙃 

Someday we will take care of the whole `let Some(opamp) else` thing...

Reply:
yes and in this case the warn message doesn't make sense, it only appears whenver opamp is enabled and sig verificatino is disabled